### PR TITLE
Add calamine Excel reader

### PR DESCRIPTION
**Pull request by davetapley**
Since https://github.com/pandas-dev/pandas/pull/54998

No existing tests covering `engine`, I can add if needed?

### CODE REVIEW

**Comment by Dr-Irv** [comment]
Thanks for the PR.  Would like opinion of @twoertwein to determine if we should add tests for `engine` keyword.  On the one hand, it would be nice to have for completeness.  On the other hand, we'd have to add a build dependency for `calamine`, which might be fine to do assuming all the installation issues work out for developers and in CI.

File: pandas-stubs/io/excel/_base.pyi [hunk 1]
```diff
@@ -9,6 +9,7 @@ from types import TracebackType
 from typing import (
     Any,
     Literal,
+    TypeAlias,
```
Comment (Dr-Irv) [inline pandas-stubs/io/excel/_base.pyi:12]: You'll need to import `TypeAlias` from `typing_extensions` so we can support earlier versions of python.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Comment by Dr-Irv** [comment]
Also, CI is failing because a change was made in pyright 1.1.356, so you can lock down pyright to 1.1.354 in this PR in `pyproject.toml`.  There's another PR out there that will allow us to upgrade pyright to 1.1.356

**Comment by twoertwein** [comment]
>  we'd have to add a build dependency for `calamine`, which might be fine to do assuming all the installation issues work out for developers and in CI.

I'm not familiar with how python-calamine is packaged. Assuming its pypi package contains everything to test it, I don't see a reason not to make it a development dependency. If we have to install packages that are not on Pypi, I would be inclined to have tests inside a TYPE_CHECKING block instead.

**Comment by Dr-Irv** [comment]
> > we'd have to add a build dependency for `calamine`, which might be fine to do assuming all the installation issues work out for developers and in CI.
> 
> I'm not familiar with how python-calamine is packaged. Assuming its pypi package contains everything to test it, I don't see a reason not to make it a development dependency. If we have to install packages that are not on Pypi, I would be inclined to have tests inside a TYPE_CHECKING block instead.

It is on pypi and conda-forge, so I think we'd be OK.  I was concerned that it is built on top of rust, and I don't know if that means a whole lot of other stuff gets installed as a result.

@davetapley Open to have you add tests to this PR

File: pyproject.toml [hunk 2]
```diff
@@ -39,7 +39,7 @@ mypy = "1.9.0"
 pandas = "2.2.1"
 pyarrow = ">=10.0.1"
 pytest = ">=7.1.2"
-pyright = ">=1.1.354"
+pyright = "=1.1.354"
```
Comment (Dr-Irv) [inline pyproject.toml:42]: #897 fixed this issue, so you can use `pyright = ">=1.1.357"` and CI should pass

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Can you merge with `upstream/main` - CI is failing because of that.  And update the pyright version as indicated.

File: pandas-stubs/io/excel/_base.pyi [hunk 3]
```diff
@@ -32,6 +35,8 @@ from pandas._typing import (
     WriteExcelBuffer,
 )
 
+ReadEngine: TypeAlias = Literal["xlrd", "openpyxl", "odf", "pyxlsb", "calamine"]
```
Comment (twoertwein) [inline pandas-stubs/io/excel/_base.pyi:38]: Could go in `pandas._typing`.

**Review by twoertwein — APPROVED** [review]

**Pull request merged**